client/wsclient_test.go

```diff
@@ -93,6 +93,62 @@ func TestWSSenderReportsHeartbeat(t *testing.T) {
 	}
 }
 
+func TestWSClientStartWithHeartbeatInterval(t *testing.T) {
+	tests := []struct {
+		name                  string
+		clientEnableHeartbeat bool
+		expectHeartbeats      bool
+	}{
+		{"client enable heartbeat", true, true},
+		{"client disable heartbeat", false, false},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			srv := internal.StartMockServer(t)
+
+			var conn atomic.Value
+			srv.OnWSConnect = func(c *websocket.Conn) {
+				conn.Store(c)
+			}
+			var msgCount atomic.Int64
+			srv.OnMessage = func(msg *protobufs.AgentToServer) *protobufs.ServerToAgent {
+				msgCount.Add(1)
+				return nil
+			}
+
+			// Start an OpAMP/WebSocket client.
+			heartbeatSec := 1
+			settings := types.StartSettings{
+				OpAMPServerURL:          "ws://" + srv.Endpoint,
+				HeartbeatIntervalSecond: &heartbeatSec,
+			}
+			if tt.clientEnableHeartbeat {
+				settings.Capabilities = protobufs.AgentCapabilities_AgentCapabilities_ReportsHeartbeat
+			}
+			client := NewWebSocket(nil)
+			startClient(t, settings, client)
+
+			// Wait for connection to be established.
+			eventually(t, func() bool { return conn.Load() != nil })
+
+			if tt.expectHeartbeats {
+				assert.Eventually(t, func() bool {
+					return msgCount.Load() >= 2
+				}, 3*time.Second, 100*time.Millisecond)
+			} else {
+				assert.Never(t, func() bool {
```

Comment:
I think this case is going to block the test for 3 seconds, waiting for a heartbeat that never arrives. We don't want to extend the duration of tests for marginal gain.

If there is no way to shorten this test I would prefer not to have it. This is a minor test case and I think the increase in test suite duration is not worth it. One possible approach is to perhaps keep the code for this case, but skip it normally. If there is a need to re-test this feature in the future we can remove the Skip() temporarily.

Same for the http client tests.

Reply:
I've reduced the test duration to 50 ms for this test case. If this isn't sufficient, we could consider removing the `expectHeartbeats == false` test case.